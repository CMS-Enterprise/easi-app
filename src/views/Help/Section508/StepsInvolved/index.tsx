import React from 'react';
import { useTranslation } from 'react-i18next';

import AccessibilityTestingSteps from 'components/AccessibilityTestingSteps';
import HelpBreadcrumb from 'components/HelpBreadcrumb';
import HelpPageIntro from 'components/HelpPageIntro';
import MainContent from 'components/MainContent';
import RelatedArticles from 'components/RelatedArticles';

const StepsInvolved = () => {
  const { t } = useTranslation('accessibility');
  return (
    <>
      <MainContent className="grid-container margin-bottom-3">
        <HelpBreadcrumb type="Close tab" />
        <HelpPageIntro
          heading={t('accessibility:testingStepsOverview.heading')}
          subheading={t('accessibility:testingStepsOverview.description')}
          type="Section 508"
        />
        <h2 className="margin-bottom-2">
          {t('testingStepsOverview.subheading')}
        </h2>
        <AccessibilityTestingSteps helpArticle />
      </MainContent>
      <RelatedArticles type="Section 508" currentArticle="stepsInvolved508" />
    </>
  );
};

export default StepsInvolved;
