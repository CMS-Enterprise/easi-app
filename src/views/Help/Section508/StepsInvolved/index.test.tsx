import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';

import StepsInvolved from './index';

describe('Section 508 Steps Involved', () => {
  it('matches the snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <StepsInvolved />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
