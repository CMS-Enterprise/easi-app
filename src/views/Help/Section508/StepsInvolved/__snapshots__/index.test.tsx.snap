// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section 508 Steps Involved matches the snapshot 1`] = `
<DocumentFragment>
  <main
    class="easi-main-content grid-container margin-bottom-3"
    id="main-content"
    role="main"
    tabindex="-1"
  >
    <button
      class="usa-button usa-button--unstyled margin-top-6"
      data-testid="button"
      type="button"
    >
      <svg
        class="usa-icon margin-right-05 margin-top-3px text-tbottom"
        focusable="false"
        height="1em"
        role="img"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
      Close tab
    </button>
    <div
      class="help-page-intro margin-bottom-4"
    >
      <h1
        aria-live="polite"
        class="easi-h1 margin-bottom-0 margin-top-3"
        tabindex="-1"
      >
        Steps involved in 508 testing
      </h1>
      <a
        class="usa-link display-inline-block margin-y-1"
        href="/help/section-508"
      >
        <span
          class="easi-tag system-profile__tag text-primary-dark bg-primary-lighter padding-bottom-1"
          data-testid="tag"
        >
          Section 508
        </span>
      </a>
      <div
        class="font-body-lg margin-top-1 margin-bottom-0 line-height-body-5"
      >
        This is an overview of the process for completing 508 testing for your system.
      </div>
    </div>
    <h2
      class="margin-bottom-2"
    >
      Steps in the testing process
    </h2>
    <div
      class="508-steps-involved"
    >
      <div
        class="easi-collapsable-link"
      >
        <button
          aria-controls="easi-508-testing-exception"
          aria-expanded="false"
          class="usa-button usa-button--unstyled display-flex flex-align-center display-flex flex-align-center margin-bottom-4"
          data-testid="collapsable-link"
          type="button"
        >
          <svg
            class="usa-icon margin-right-05"
            focusable="false"
            height="1em"
            role="img"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
            />
          </svg>
          What if I need an exception from 508 testing?
        </button>
      </div>
      <ol
        class="usa-process-list"
      >
        <li
          class="usa-process-list__item grid-col-8 position-static maxw-desktop"
        >
          <h3
            class="usa-process-list__heading"
            data-testid="processListHeading"
          >
            Fill out the request form in EASi
          </h3>
          <p>
            Tell the 508 team which system you plan to test.
          </p>
        </li>
        <li
          class="usa-process-list__item grid-col-8 position-static maxw-desktop"
        >
          <h3
            class="usa-process-list__heading"
            data-testid="processListHeading"
          >
            Prepare and upload the VPAT and Test plan
          </h3>
          Download and fill the VPAT and Test plan from the 
          <a
            class="usa-link"
            href="/508/templates"
            rel="noopener noreferrer"
            target="_blank"
          >
            templates page (opens in a new tab)
          </a>
          . These documents will help the 508 team prepare for testing. Upload your completed documents to EASi for the 508 team to review.
          <p
            class="margin-bottom-0"
          >
            The 508 team will get back to you via email about any changes needed prior to testing.
          </p>
        </li>
        <li
          class="usa-process-list__item grid-col-8 position-static maxw-desktop"
        >
          <h3
            class="usa-process-list__heading"
            data-testid="processListHeading"
          >
            Attend the testing session
          </h3>
          <p>
            The 508 team will work with you to schedule a testing session and you will test your application together. Depending on the results, you may need to address any issues and retest.
          </p>
        </li>
        <li
          class="usa-process-list__item grid-col-8 position-static maxw-desktop"
        >
          <h3
            class="usa-process-list__heading"
            data-testid="processListHeading"
          >
            Receive results
          </h3>
          <p>
            Your test score will determine your next steps:
          </p>
          <dl
            title="508 test scores"
          >
            <dt
              class="text-bold display-flex flex-align-center"
            >
              <svg
                class="usa-icon height-3 width-3 margin-right-1 text-success"
                focusable="false"
                height="1em"
                role="img"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0V0zm0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M16.59 7.58 10 14.17l-3.59-3.58L5 12l5 5 8-8zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                />
              </svg>
              99% and above
            </dt>
            <dd
              class="margin-left-4 margin-bottom-2"
            >
              508 testing is complete and you can release your application.
            </dd>
            <dt
              class="text-bold display-flex flex-align-center"
            >
              <svg
                class="usa-icon height-3 width-3 margin-right-1 text-warning-dark"
                focusable="false"
                height="1em"
                role="img"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                />
              </svg>
              Between 99% and 75%
            </dt>
            <dd
              class="margin-left-4 margin-bottom-2"
            >
              You can release your application but you will need fix the issues and retest within a year. You need to submit a remediation plan as a part of retesting. You can download the Remediation Plan from the 
              <a
                class="usa-link"
                href="/508/templates"
                target="_blank"
              >
                templates page (opens in a new tab)
              </a>
              .
            </dd>
            <dt
              class="text-bold display-flex flex-align-center"
            >
              <svg
                class="usa-icon height-3 width-3 margin-right-1 text-error"
                focusable="false"
                height="1em"
                role="img"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M14.59 8 12 10.59 9.41 8 8 9.41 10.59 12 8 14.59 9.41 16 12 13.41 14.59 16 16 14.59 13.41 12 16 9.41 14.59 8zM12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                />
              </svg>
              75% and below
            </dt>
            <dd
              class="margin-left-4"
            >
              You cannot release your application. You need to fix all issues right away and retest.
            </dd>
          </dl>
        </li>
      </ol>
    </div>
  </main>
  <div
    class="bg-base-lightest"
  >
    <main
      class="easi-main-content grid-container padding-y-2"
      id="main-content"
      role="main"
      tabindex="-1"
    >
      <h2
        class="margin-bottom-1"
      >
        Related help articles
      </h2>
      <dt
        class="margin-bottom-4"
      >
        Below are some additional help articles that you may find useful.
      </dt>
      <ul
        class="usa-card-group margin-y-2"
        data-testid="CardGroup"
      >
        <li
          class="usa-card desktop:grid-col-4 article article-card--isLink"
          data-testid="article-card"
        >
          <div
            class="usa-card__container radius-md shadow-2 minh-mobile"
          >
            <header
              class="usa-card__header padding-x-3 padding-top-3"
              data-testid="CardHeader"
            >
              <h3
                class="line-height-body-4 margin-bottom-1"
              >
                Templates for 508 Testing
              </h3>
              <a
                class="usa-link display-inline-block margin-bottom-1"
                href="/help/section-508"
              >
                <span
                  class="easi-tag system-profile__tag text-primary-dark bg-primary-lighter padding-bottom-1"
                  data-testid="tag"
                >
                  Section 508
                </span>
              </a>
            </header>
            <div
              class="usa-card__body padding-top-0 article__body"
              data-testid="CardBody"
            >
              <p>
                Download and fill out these templates as needed to complete 508 testing for your system
              </p>
            </div>
            <div
              class="usa-card__footer padding-top-2 article__footer"
              data-testid="CardFooter"
            >
              <a
                class="usa-link easi-request__button-link padding-x-2"
                href="/"
              >
                Read
              </a>
            </div>
          </div>
        </li>
      </ul>
    </main>
  </div>
</DocumentFragment>
`;
